Add fetchSubjects effect

jotnguyen opened the pull request:
Added fetchSubjects effect to browse. Currently trying to implement the action and reducer

Commit:
Unscheduled Classes Box

jotnguyen committed:
Add fetchSubjects effect

jotnguyen committed:
Add fetchSubjects effect

jotnguyen committed:
Add fetchSubjects effect

jotnguyen commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Implement fetchSubjects effect](https://trello.com/c/KbW5Tgzv/73-implement-fetchsubjects-effect)

Joonpark13 commented on src/effects/browse.js:
```diff
@@ -12,3 +12,14 @@ export function fetchSchools() {
+export function fetchSubjects(termId, subjectId) {
+  return currentTermDoc
+    .collection('subjects')
+    .where('termId', '==', termId)
```
You're not gonna need this line since this subjects subcollection is already inside the appropriate term document (`currentTermDoc`). This _does_ mean that you won't be using the argument `termId` for now but that's okay since when we implement multiple terms we will be using this.

jotnguyen replied:
So if I understand your comment correctly, subjects is under currentTermDoc so .collection('subjects') is fine and the currentTerm is already matched because of db.collection('terms').doc(CURRENT_TERM). I can just remove the first .where()

Commit:
Cloud Firestore Security

Add auth to follow the newly implemented cloud firestore security rules.

Commit:
Size classes

Adjust the height of the classes in the calendar to match the length of
the class.

jotnguyen committed:
Remove termId from subjects query

jotnguyen committed:
Remove termId from subjects query

jotnguyen committed:
Remove termId from subjects query

Joonpark13 commented on src/effects/browse.js:
```diff
@@ -12,3 +12,13 @@ export function fetchSchools() {
+export function fetchSubjects(termId, subjectId) {
+  return currentTermDoc
+    .collection('subjects')
+    .where('subjectId', '==', subjectId)
```
this should be `.where(`id`, '==', subjectId)` since for the subjects objects, the identifier will just be `id`. See https://github.com/Joonpark13/sans-serif#subject

Commit:
Style classes

Each class on the calendar is now styled, and displays the time, course
id, section name, and has a color. The colors are drawn from the
Northwestern Web Colors and rotate through a list of 12.

jotnguyen committed:
Change searched attribute from 'subjectId' to id

jotnguyen committed:
Change searched attribute from 'subjectId' to id

jotnguyen committed:
Change searched attribute from 'subjectId' to id

jotnguyen committed:
Change id to 'id'

jotnguyen committed:
Change id to 'id'

jotnguyen committed:
Change id to 'id'

Commit:
Style overlapping classes

Joonpark13 commented on src/effects/browse.js:
```diff
@@ -12,3 +12,13 @@ export function fetchSchools() {
+
+export function fetchSubjects(termId, subjectId) {
+  return currentTermDoc
+    .collection('subjects')
```
You're gonna need to still make sure the subject is associated with the correct school as well, since some subjects can be in multiple schools.

jotnguyen committed:
Modify fetchSubjects query so it gets correct subjects for term and school

jotnguyen committed:
Merge branch 'f/fetch-subjects' of https://github.com/Joonpark13/serif.nu into f/fetch-subjects
